kpatch-build/create-diff-object.c

```diff
@@ -2833,7 +2859,9 @@ int main(int argc, char *argv[])
 
 	list_for_each_entry(sym, &kelf_base->symbols, list) {
 		if (sym->type == STT_FILE) {
-			hint = sym->name;
+			hint = strdup(sym->name);
+			if (!hint)
+				ERROR("failed to copy file name.");
```

Comment:
I think we can take a different approach here.  The hint is actually no longer needed in kpatch_create_patches_sections() and kpatch_create_intermediate_sections().  It _used_ to be needed for the lookup_local_symbol() calls, but that code was restructured and we forgot to update the error messages accordingly.  The error messages no longer need to show the hint.

Reply:
So, we can get rid of that 'hint' altogether. Good.

Reply:
> So, we can get rid of that 'hint' altogether. Good.

Ah, I see. `hint` is still needed for `lookup_open()` and such but it is no longer necessary after `kelf_base` is freed.